mmdet3d/core/bbox/coders/monoflex_bbox_coder.py

```diff
@@ -0,0 +1,387 @@
+import numpy as np
+import torch
+from torch.nn import functional as F
+
+from mmdet.core.bbox import BaseBBoxCoder
+from mmdet.core.bbox.builder import BBOX_CODERS
+
+
+@BBOX_CODERS.register_module()
+class MonoFlexCoder(BaseBBoxCoder):
+    """Bbox Coder for MonoFlex.
+
+    Args:
+        depth_mode (str): The depth mode for direct depth calculation.
+        base_depth (tuple[float]): Depth references for decode box depth.
+        depth_range (str): Depth range of predicted depth.
+        use_combined_depth (bool): Whether to use combined depth (direct depth
+            and depth from keypoints).
+        dims_mean (tuple[tuple[float]]): Dimensions mean of decode bbox
+            dimensions [l, h, w] for each category.
+        dims_mean (tuple[tuple[float]]): Dimensions std of decode bbox
+            dimensions [l, h, w] for each category.
+        dims_modes (list[str|bool]): Dimensions modes.
+        multibin (bool): Whether to use multi_bin representation.
+        alpha_centers (list[float]): Alpha centers while using multi_bin
+            representations.
+        num_dir_bin (int): Number of Number of bins to encode
+            direction angle.
+        code_size (int): The dimension of boxes to be encoded.
+    """
+
+    def __init__(
+        self,
+        depth_mode,
+        base_depth,
+        depth_range,
+        use_combined_depth,
+        dims_mean,
+        dims_std,
+        dims_modes,
+        multibin,
+        alpha_centers,
+        num_dir_bin,
+        code_size,
+    ):
+        super(MonoFlexCoder, self).__init__()
+
+        # depth related
+        self.depth_mode = depth_mode
+        self.base_depth = base_depth
+        self.depth_range = depth_range
+        self.use_combined_depth = use_combined_depth
+
+        # dimensions related
+        self.dims_mean = dims_mean
+        self.dims_std = dims_std
+        self.dims_modes = dims_modes
+
+        # orientation related
+        self.multibin = multibin
+        self.alpha_centers = alpha_centers
+        self.num_dir_bin = num_dir_bin
+
+        # output related
+        self.bbox_code_size = code_size
+        self.eps = 1e-3
+
+    def encode(self, locations, dimensions, orientations, input_metas):
+        """Encode CameraInstance3DBoxes by locations, dimensions, orientations.
+
+        Args:
+            locations (Tensor): Center location for 3D boxes.
+                (N, 3)
```

Comment:
Center location for 3D boxes of shape (N,3).
Please also unify the format of the following doc of arguments.